tests/accessor/generic_accessor_api_common.h

```diff
@@ -438,13 +438,24 @@ class run_api_tests {
   }
 };
 
-template <typename T>
+using test_combinations =
+    typename get_combinations<access_modes_pack, all_dimensions_pack,
+                              targets_pack>::type;
+
+template <typename T, typename ArgCombination>
 class run_generic_api_for_type {
  public:
   void operator()(const std::string &type_name) {
-    const auto access_modes = get_access_modes();
-    const auto dimensions = get_all_dimensions();
-    const auto targets = get_targets();
+    // Get the packs from the test combination type.
+    using AccessModePack = typename std::tuple_element<0, ArgCombination>::type;
+    using DimensionsPack = typename std::tuple_element<1, ArgCombination>::type;
+    using TargetsPack = typename std::tuple_element<2, ArgCombination>::type;
```

Comment:
While you are here, you could change everywhere
```suggestion
    using AccessModePack = typename std::tuple_element_t<0, ArgCombination>;
    using DimensionsPack = typename std::tuple_element_t<1, ArgCombination>;
    using TargetsPack = typename std::tuple_element_t<2, ArgCombination>;
```

Reply:
Absolutely! It has been done. 👍